### DISCUSSION TITLE
Any guide to make generic webhook work with discord?

### DISCUSSION THREAD

**Discussion by dragonnest21** [sct/overseerr#1074] [Help] [answered]
First pardon my english. Not really good at it. 

Anyway the setup looks pretty straight forward( or so i thought)
Below is the default setup. I just added my discord webhook. Hopefully someone can shed light. Played a little with 
the JSON payload.

Yet it won't send any notification in discord. I know overseerr supports discord notification but i would like to
customize the notification sent to discord. I want to add a little instruction like to wait for a few minutes and that their 
request has been accepted and is currently being searched. It is the only reason why i have not completely changed to overserr yet. But would really love to as i find the UI and overall experience better.

![Screenshot_2021-03-04 Settings - Overseerr(1)](https://user-images.githubusercontent.com/79920630/109845238-e0dde600-7c87-11eb-9e81-45d0ed53d14f.png)

**Comment by TheCatLady** [comment, marked as answer]
You need to follow the Discord API, but we do not officially support this.

**Reply by dragonnest21** [reply]
> 
> 
> You need to follow the Discord API, but we do not officially support this.

Yes i was following the discord API. still cant make it work. Hopefully it gets added in the discord notification. Was just hoping someone with knowledge about it would reply.

**Reply by TheCatLady** [reply]
@dragonnest21 The JSON payload in your screenshot does not follow the Discord API spec.

**Reply by TheCatLady** [reply]
The way to do this is to create a webhook (I personally like adnanh/webhook) and do your processing there.

**Reply by dragonnest21** [reply]
> 
> 
> @dragonnest21 The JSON payload in your screenshot does not follow the Discord API spec.

Oh wow. This actually solved my problem. I thought JSON payload/format is sort of universal. Did not know discord has its own. Thank you! Now time to setup reverse proxy so i can use my own custom domain :)

**Reply by EL-R1** [reply]
Do you still have all your configurations ? and how you did it ?

**Reply by grenius** [reply]
@EL-R1 This is what I use for notifying for new requests. I haven't been able to make one big json with everything I want yet, but here you go.

```
{
    "embeds": [
        {
            "color": 16744468,
            "author": {
                "name": "New media request"
            },
            "title": "{{subject}}",
            "url": "https://MY_OVERSEER_EXT_URL/{{media_type}}/{{media_tmdbid}}",
            "description": "{{message}}",
            "thumbnail": {
                "url": "{{image}}"
            },
            "fields": [
                {
                    "name": "Requested by",
                    "value": "{{requestedBy_username}}",
                    "inline": true
                },
                {
                    "name": "Status",
                    "value": "{{media_status}}",
                    "inline": true
                }
            ]
        }
    ]
}
```

**Reply by EL-R1** [reply]
> @EL-R1 This is what I use for notifying for new requests. I haven't been able to make one big json with everything I want yet, but here you go.
> 
> ```
> {
>     "embeds": [
>         {
>             "color": 16744468,
>             "author": {
>                 "name": "New media request"
>             },
>             "title": "{{subject}}",
>             "url": "https://MY_OVERSEER_EXT_URL/{{media_type}}/{{media_tmdbid}}",
>             "description": "{{message}}",
>             "thumbnail": {
>                 "url": "{{image}}"
>             },
>             "fields": [
>                 {
>                     "name": "Requested by",
>                     "value": "{{requestedBy_username}}",
>                     "inline": true
>                 },
>                 {
>                     "name": "Status",
>                     "value": "{{media_status}}",
>                     "inline": true
>                 }
>             ]
>         }
>     ]
> }
> ```

Thx man,
I figure it out on my own.
If anyone wants a sonarr-overseer-webhook setup let me know
